src/collections/unordered-map.ts

```diff
@@ -128,32 +128,32 @@ export class UnorderedMap {
     this.values.clear();
   }
 
-  toArray(): [Bytes, unknown][] {
+  toArray(): [Bytes, T][] {
     let ret = [];
     for (let v of this) {
       ret.push(v);
     }
     return ret;
   }
 
-  [Symbol.iterator](): UnorderedMapIterator {
+  [Symbol.iterator](): UnorderedMapIterator<T> {
     return new UnorderedMapIterator(this);
   }
 
-  extend(kvs: [Bytes, unknown][]) {
+  extend(kvs: [Bytes, T][]) {
     for (let [k, v] of kvs) {
       this.set(k, v);
     }
   }
 
   serialize(): string {
-    return JSON.stringify(this)
+    return JSON.stringify(this);
   }
 
   // converting plain object to class object
-  static deserialize(data: UnorderedMap): UnorderedMap {
+  static deserialize(data: UnorderedMap<unknown>): UnorderedMap<unknown> {
```

Comment:
This should be a static function with a header something like this: 
```typescript
static deserialize<DataType>(data: UnorderedMap<DataType>): UnorderedMap<DataType> {
```

Reply:
Not sure what you mean. Why you are not using T here?

Reply:
Right now this function is not generic amd will return an unknown. I am using `DataType` instead of `T` because that is just the generic name, you can use anything for `T` I just like using more descriptive names as it is self documenting for the library users

Reply:
:+1: